internal/provider/client_ca_cert_resource.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2022 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	sdk_resource "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	clientCACertEndpointTimeout = time.Minute * 10 // TODO: what's a reasonable value here?
+)
+
+type clientCACertResource struct {
+	provider *provider
+}
+
+func NewClientCACertResource() resource.Resource {
+	return &clientCACertResource{}
+}
+
+func (r *clientCACertResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages client CA certs.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
```

Comment:
I'm using the `id` here as the `cluster_id`, is that ok?  The idea is that the cluster_id always need to be provided, so the `id` isn't a computed, "known after apply" field (since the client CA cert feature doesn't really have a concept of it's own id)